modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/GridCoverageRenderer.java

```diff
@@ -1164,8 +1178,14 @@ public void paint(final Graphics2D graphics, final GridCoverage2DReader gridCove
 
         setupInterpolationHints(interpolation);
         
+        //
+        // Band selection
+        //
+        applyBandsSelectionParameterIfNeeded(gridCoverageReader, readParams, symbolizer);
+        RasterSymbolizer finalSymbolizer = setupSymbolizerIfBandsSelectionIsUsed(symbolizer);
```

Comment:
Fixed in [54bdacb](https://github.com/mantesat/geotools/commit/54bdacb58e7edb75639ec16aac993c132b7f7c90)